include/boost/predef/compiler/nvcc.h

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright Benjamin Worpitz 2018
+Distributed under the Boost Software License, Version 1.0.
+(See accompanying file LICENSE_1_0.txt or copy at
+http://www.boost.org/LICENSE_1_0.txt)
+*/
+
+#ifndef BOOST_PREDEF_COMPILER_NVCC_H
+#define BOOST_PREDEF_COMPILER_NVCC_H
+
+#include <boost/predef/version_number.h>
+#include <boost/predef/make.h>
+
+/*`
+[heading `BOOST_COMP_NVCC`]
+
+[@https://en.wikipedia.org/wiki/NVIDIA_CUDA_Compiler NVCC] compiler.
+Version number available as major, minor, and patch beginning with version 7.5.
+
+[table
+    [[__predef_symbol__] [__predef_version__]]
+
+    [[`__NVCC__`] [__predef_detection__]]
+
+    [[`__CUDACC_VER_MAJOR__`, `__CUDACC_VER_MINOR__`, `__CUDACC_VER_BUILD__`] [V.R.P]]
+    ]
+ */
+
+#define BOOST_COMP_NVCC BOOST_VERSION_NUMBER_NOT_AVAILABLE
+
+#if defined(__NVCC__)
+#   if !defined(__CUDACC_VER_MAJOR__) || !defined(__CUDACC_VER_MINOR__) || !defined(__CUDACC_VER_BUILD__)
+#       define BOOST_COMP_NVCC_DETECTION BOOST_VERSION_NUMBER_AVAILABLE
+#   else
+#       define BOOST_COMP_NVCC_DETECTION BOOST_VERSION_NUMBER(__CUDACC_VER_MAJOR__, __CUDACC_VER_MINOR__, __CUDACC_VER_BUILD__)
```

Comment:
I am not an expert on `BOOST_VERSION_NUMBER` but is this save? It looks to me like the numbers are just stitched together (concatination):
https://github.com/boostorg/predef/blob/822d09f19bc2f4ea6f42da8e0be83d10ce912ce1/test/version.cpp#L32

Released CUDA versions have currently the length:
X.Y.ZZ
and
X.Y.ZZZ

Reply:
here is how we stitch it together in boost::config
https://github.com/boostorg/config/pull/175/files

Reply:
It should be safe. See [here](https://github.com/boostorg/predef/blob/develop/include/boost/predef/version_number.h#L36). The multiplication is nearly the same as the config version but it seems to have one more patch digit.